release-notes.txt

```diff
@@ -1,3 +1,565 @@
+* pkg/driver_cryptocell_310: allow data to be in ROM on hash update (#21214)
+* pkg/driver_cryptocell_310: check for input data to be in RAM (#21138)
+* pkg/esp32_sdk: Fix broken `rtc_io.h` declaration and prevent
+  `sys/uio.h` inclusion outside RIOT (#21408)
+
+Boards
+------
+
++ Add support for Seeed Studio ESP32C3 Xiao (#21267)
++ boards/feather-nrf52840*: add external flash configuration (#21324)
++ boards/nucleo-f722ze: add ADC support (#21233)
++ boards/nucleo-wl55jc: add ADC configuration (#21235)
++ boards/nucleo64: Add Compile Warning about LED0 when using SPI (#21338)
++ boards/seeedstudio-xiao-nrf52840-sense: Add Initial Support (#21332)
++ boards: introduce `stdio_default` (#21294)
++ boards: introduce board alias and make native default to host target (#21242)
++ Buildsystem: Introduce Global Makefiles for `boards` Directory (#21327)
++ cpu/sam4s: add initial support with SAM4S-XPRO board (#21201)
+* boards/common: Make Adafruit nRF52 Bootloader shared (#21281)
+* boards/feather-nrf52840*: rename to adafruit-feather-nrf52840-* (#21349)
+* boards/nrf6310: Remove nRF6310 board from RIOT (#20966)
+* boards/seeedstudio-xiao-nrf52840: initial board support (#20980)
+
+CPU
+---
+
++ core: add stdio.h to replace stdout functions with stdio_null (#20872)
+* cpu/native: Gardening/QoL (#21283)
+* cpu/nrf5*: mark and resolve radio conflict (#21250)
+* cpu/nrf52/radio/nrf802154: fix beacon acceptance (#20982)
+* cpu/sam0_common/flashpage: disable interrupts while writing (#21326)
+* cpu/sam0_eth: fix hang with broken PHY (#21172)
+* cpu/sam3: fix hwrng peripheral register access (#21177)
+* cpu/sam3: optimize gpio ISR processing (#21179)
+* cpu/samd5x: improve can-initialization (#21173)
+* cpu/stm32{f0,g0,c0}: fix ADC initialization sequence (#21230)
+* dist/tools/esptools: Upgrade the ESP32 toolchain to GCC v14.2, GDB
+  v14.2 and OpenOCD v0.12 (#21144)
+* {cpu/sam0_common/sam0_sdhc,drivers/mtd}: do not allocate `work_area`
+  twice (#21169)
+
+Device Drivers
+--------------
+
++ drivers/periph_timer: add `timer_get_closest_freq()` (#20581)
++ drivers/sen5x: Add device driver for SEN5x (#19955)
++ drivers/tm1637: add driver and tests for tm1637 7-segment display (#21112)
+* driver/sen5x: Small Fixups (#21392)
+* drivers/at: at_readline_stop_at_str fix remaining buffer length (#21195)
+* drivers/sdmmc/sdmmc_sdhc: remove DMA check (#21260)
+* drivers/{mtd_sdmmc,sdmmc}: reinit card (#21305)
+
+Documentation
+-------------
+
++ add *.doc.md to doxygen sources (#21319)
++ boards/nrf52840{dk,dongle}: add MCU table, move to doc.md (#21271)
++ boards/nucleo-f413zh: add MCU table, move to doc.md (#21264)
++ dist/tools: Add Doxygen as a Tool for CI (#21300)
++ doc, buildsystem: Add a Version Check for Doxygen, Fix doc* Target
+  Execution (#21277)
++ doc/managing-a-release: add looking out for release tests to preps (#21368)
++ doc: add documentation on RIOT_FILE_RELATIVE (#21170)
++ doc: add psutil requirement to  Getting Started (#21399)
```

Comment:
This is also something we might want to point out in the cover letter that `psutil` is now needed for `pyterm`.